### PR TITLE
ZomeCall expects base64

### PR DESCRIPTION
**Pull request by zo-el**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -428,29 +428,26 @@ class Envoy {
...
-        // In case of no call args, convert empty obj to null
-        if (Object.keys(call_spec.args).length <= 0) {
-          log.debug('No call_spec.args, converting value to null for zomeCall.');
-          call_spec.args = null
-        };
-
         const hosted_app_cell_id = call_spec["cell_id"];
+        let payload = call_spec["args"]
+        log.silly('Original payload for zomeCall:', payload);
+        if (typeof payload === 'string') {
```
Comment (timotree3) [inline src/index.ts]: When would payload not be a `string`?

File: src/index.ts [hunk 2]
```diff
@@ -428,29 +428,26 @@ class Envoy {
...
+          payload = msgpack.decode(Buffer.from(call_spec["args"], 'base64'));
```
Comment (timotree3) [inline src/index.ts:434]: I would prefer to use a new variable rather than changing the type of an existing variable (from `string` to `any`)

**Review by timotree3 — APPROVED** [review]
Looks like it would work as is so feel free to merge. I left a question and a suggestion

File: src/index.ts [hunk 3]
```diff
@@ -428,29 +428,26 @@ class Envoy {
...
-
-        // In case of no call args, convert empty obj to null
-        if (Object.keys(call_spec.args).length <= 0) {
-          log.debug('No call_spec.args, converting value to null for zomeCall.');
-          call_spec.args = null
-        };
-
         const hosted_app_cell_id = call_spec["cell_id"];
+        let payload = call_spec["args"]
+        log.silly('Original payload for zomeCall:', payload);
```
Comment (timotree3) [inline src/index.ts]: I don't think we should have this log. It will just ouput a bunch of useless base64

File: src/index.ts (hunk 1 above)
Comment (zo-el) [reply src/index.ts]: That's only if the payload was not passed

**Review by timotree3 — APPROVED** [review]

**Review by timotree3 — APPROVED** [review]

**Pull request merged**